src/Plugin/Commerce/PaymentGateway/Heartland.php

```diff
@@ -231,8 +231,18 @@ public function createPayment(PaymentInterface $payment, $capture = true)
                 $response = $response->execute();
 
                 // Updating the payment method to contain the multi-use token for next time.
-                $payment_method->setRemoteId('mut'.$response->token);
-                $payment_method->save();
+                if (isset($response->token) && !empty($response->token)) {
+                    $payment_method->setRemoteId('mut'.$response->token);
+                    $payment_method->setReusable(true);
+                    $payment_method->save();                                 
+                } else {
+                    // This has to be set in createPaymentMethod so Drupal will save non-sensitive card info
+                    // We're going to try to unset it here since retrieving the token failed
+                    $payment_method->setReusable(false);
+                    $payment_method->save();
+                    throw new PaymentGatewayException('There was an issue retrieving multi-use token. Response Code: '.$response->responseCode.' Response Message: '.$response->responseMessage);
```

Comment:
Would this fail the transaction on the Drupal side? If so, we should remove this to let storing the card within Drupal fail silently since the payment itself was successful.

Reply:
Yep, not having multi-use tokenization is the typical cause. We're handling this in other platforms in the same way. If the token isn't in the response, we don't show the error anywhere and don't store anything on the customer's account.